### ISSUE TITLE
C compiler warning about void* dereferencing

### ISSUE THREAD

**Issue by jakirkham** [rapidsai/ucx-py#125]
When building ucx-py, the Cythonized C code contains a line (guessing it maps to [this one]( https://github.com/rapidsai/ucx-py/blob/4030eff63438ac87d58fd009656198a184914041/ucp/_libs/ucp_py_buffer_helper.pyx#L156 )), which the compiler warns is dereferencing a `void*`. Have included a brief snippet of the warning below.

```c
ucp/_libs/ucp_py_buffer_helper.c: In function '__pyx_pf_20ucp_py_buffer_helper_12BufferRegion_4__getbuffer__':
ucp/_libs/ucp_py_buffer_helper.c:3472:61: warning: dereferencing 'void *' pointer
     __pyx_v_buffer->buf = ((void *)(&(__pyx_v_self->buf->buf[0])));
```

**Comment by madsbk** [comment]
Fixed in #144

**Comment by jakirkham** [comment]
Awesome! Thanks Mads 😄
